rmf_traffic/test/unit/agv/test_Planner.cpp

```diff
@@ -2855,277 +2856,628 @@ SCENARIO("Test computing Starts from coordinates")
       std::size_t waypoint_index = start_set[i].waypoint();
       CHECK(waypoint_indices.insert(waypoint_index).second);
       CHECK((waypoint_index == 1 || waypoint_index == 2));
-      
       CHECK(start_set[i].location());
+      CHECK(start_set[i].lane());
+      std::size_t& lane_index = start_set[i].lane().value();
+      CHECK(lane_indices.insert(lane_index).second);
+      if (waypoint_index == 1)
+        CHECK(lane_index == 2);
+      else if (waypoint_index == 2)
+        CHECK(lane_index == 1);
+      else
+        CHECK(false);
+    }
+  }
+
+  WHEN("Location is not within merging with waypoints, within lane merging, "
+      "between unidirectional lane entry and exit.")
+  {
+    graph.add_waypoint(test_map_name, {0, 0}); // 0
+    graph.add_waypoint(test_map_name, {1, 0}); // 1
+    graph.add_waypoint(test_map_name, {10, 0}); // 2
+    REQUIRE(graph.num_waypoints() == 3);
+
+    graph.add_lane(0, 1); // 0
+    graph.add_lane(1, 2); // 1
+    REQUIRE(graph.num_lanes() == 2);
+
+    planner = Planner{Planner::Configuration{graph, traits}, default_options};
```

Comment:
It looks to me like this `Planner` instance only ever gets used to store the `Graph`, so it's not really accomplishing anything. It may be worth checking the other start tests to see if they are similarly using an unnecessary `Planner` instance.

Reply:
thanks for catching this!